autogpt_platform/frontend/src/components/TypeBasedInput.tsx

```diff
@@ -0,0 +1,428 @@
+import React, { FC } from "react";
+import { cn } from "@/lib/utils";
+import { format } from "date-fns";
+import { CalendarIcon } from "lucide-react";
+import { Cross2Icon, FileTextIcon } from "@radix-ui/react-icons";
+
+import { Input as DefaultInput } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import { Switch } from "@/components/ui/switch";
+import { Button } from "@/components/ui/button";
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+} from "@/components/ui/popover";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from "@/components/ui/select";
+import { determineDataType, DataType } from "@/lib/autogpt-server-api/types";
+import { BlockIOSubSchema } from "@/lib/autogpt-server-api/types";
+
+/**
+ * A generic prop structure for the TypeBasedInput.
+ *
+ * onChange expects an event-like object with e.target.value so the parent
+ * can do something like setInputValues(e.target.value).
+ */
+export interface TypeBasedInputProps {
+  schema: BlockIOSubSchema;
+  value?: any;
+  placeholder?: string;
+  onChange: (value: any) => void;
+}
+
+const inputClasses = "min-h-11 rounded-[55px] border px-4 py-2.5";
```

Comment:
`rounded-[55px]`? Shouldn't this just be `rounded-full`?

Reply:
Every single custom pixel value is tech debt

Reply:
rounded full is a circle, while rounded 55px is 3.4375rem

Reply:
are the ends not fully rounded?

Reply:
Okay updated to rounded-full